tracing-journald/src/socket.rs

```diff
@@ -31,14 +33,41 @@ fn cmsg_buffer_size_for_one_fd() {
     assert_cmsg_bufsize()
 }
 
-pub fn send_one_fd(socket: &UnixDatagram, fd: RawFd) -> Result<usize> {
+fn create_sockaddr_un_for_path<P: AsRef<Path>>(path: P) -> Result<sockaddr_un> {
+    let mut addr: sockaddr_un = unsafe { std::mem::zeroed() };
+    addr.sun_family = AF_UNIX as _;
+
+    let path_bytes = path.as_ref().as_os_str().as_bytes();
+    // We need to take the final NULL byte into account.
+    if path_bytes.len() < addr.sun_path.len() - 1 {
+        unsafe {
+            std::ptr::copy_nonoverlapping(
+                path_bytes.as_ptr(),
+                addr.sun_path.as_mut_ptr() as *mut u8,
+                path_bytes.len(),
+            );
+        }
+        addr.sun_path[addr.sun_path.len() - 1] = 0;
```

Comment:
This is unnecessary since the sockaddr_un is initially zeroed, and also confusing since it ignores the length of the path.

Reply:
Removed it.